modules/iknowpy/iknowpy/engine.pyx

```diff
@@ -41,75 +43,113 @@ cdef char* aType_to_str(Attribute t) except NULL:
 	raise ValueError('Attribute type {} is unrecognized.'.format(t))
 
 
-cdef class UserDictionary:
+Entry = namedtuple("Entry", ["literal", "label"])
+
+cdef class UserDictionary(object):
 	"""A class that represents a user dictionary"""
 	cdef CPPUserDictionary user_dictionary
 
-	def __cinit__(self):
-		"""Initialize the underlying C++ iKnowUserDictioanry class"""
+	# keep track of entries in a readable form
+	cdef public entries
+
+	def __cinit__(self, load_entries=[]):
```

Comment:
The `load_entries` parameter is unnecessary here. You don't use it in the function, and `__cinit__()` automatically ignores extra arguments when an object is constructed.